expat/CMakeLists.txt

```diff
@@ -324,6 +325,40 @@ if(BUILD_tests)
     add_test(runtestspp tests/runtestspp)
 endif(BUILD_tests)
 
+if(BUILD_fuzzers)
+    if(NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+        message(SEND_ERROR
+            "Building fuzz targets without clang is not supported. Please set "
+            "-DCMAKE_C_COMPILER=clang.")
+    endif()
+
+    string(FIND "${CMAKE_EXE_LINKER_FLAGS}" "-fsanitize" sanitizer_present)
+    if(${sanitizer_present} EQUAL "-1")
+        message(WARNING
+            "There was no sanitizer present when building the fuzz targets. "
+            "This is likely in error - consider adding "
+            "-DCMAKE_C_FLAGS='-fsanitize=<sanitizer>' and "
+            "-DCMAKE_EXE_LINKER_FLAGS='-fsanitize=<sanitizer>' to your cmake "
+            "execution.")
+    endif()
+
+    set(encoding_types UTF-16 UTF-8 ISO_8859_1 US_ASCII UTF_16BE UTF_16LE)
+    set(fuzz_targets xml_parse_fuzzer xml_parsebuffer_fuzzer)
+
+    foreach(fuzz_target ${fuzz_targets})
+        foreach(encoding_type ${encoding_types})
+            set(target_name ${fuzz_target}_${encoding_type})
+            add_executable(${target_name} fuzz/${fuzz_target}.c ${expat_SRCS})
+            target_compile_definitions(${target_name}
+                PRIVATE ENCODER_FOR_FUZZING=${encoding_type})
```

Comment:
Not the most important thing but still feels off, semantically: Can we turn "encoders" into "encodings"?